unsplahAPI/unsplahAPI/Services/UnsplashAPIService/NetworkService.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  UnsplashAPIService.swift
+//  unsplashAPI
+//
+//  Created by Никита Данилович on 25.05.2023.
+//
+
+import Network
+import UIKit
+
+enum HTTPRequstMethod: String {
+    case get = "GET"
+    case post = "POST"
+    case put = "PUT"
+    case delete = "DELETE"
+    case head = "HEAD"
+}
+
+enum HTTPResponseCode: String, Error {
+    case badRequest = "Unfortunately your request didn't find any results!"
+    case accessDenied = "Your access is denied!"
+    case notFound = "Unfortunately your request didn't find any images!"
+    case successfull
+    case invalidHTTPRequest = "Something went wrong..."
+    case invalidJSONDecoder = "Something went wrond, we are fixing it right now..."
+
+    var responseCode: Int {
+        switch self {
+        case .successfull: return 200
+        case .badRequest: return 400
+        case .accessDenied: return 403
+        case .notFound: return 404
+        case .invalidHTTPRequest: return 406
+        case .invalidJSONDecoder: return 407
+
+        }
+    }
+}
+
+final class NetworkService: NetworkServiceProtocol {
+    func startSessionForExtractingImages(
+        urlPath: String,
+        completion: @escaping (Result<Data, HTTPResponseCode>) -> Void
+    ) {
+         requestImagesFromUnsplash(requestURLPath: urlPath) { data, _, _ in
+            guard let data = data else {
+                return completion(.failure(.notFound))
+            }
+             return completion(.success(data))
+        }
+    }
+    
+    private func requestImagesFromUnsplash(
+        requestURLPath: String,
+        serverResponse: @escaping(Data?, URLResponse?, Error?) -> Void
+    ) {
+        do {
+            guard let request = try generateSplashAPIRequest(
+                for: requestURLPath,
+                httpMethod: .get
+            ) else {
+                print(HTTPResponseCode.invalidHTTPRequest)
```

Comment:
here we should return the completion in every case, or clients will be unable to make an action based on the result (in this case in ViewModel the alert won't show up)
```suggestion
                serverResponse(nil, nil, HTTPResponseCode.invalidHTTPRequest)
```

Reply:
fixed